Tasks/DownloadFileShareArtifactsV0/main.ts

```diff
@@ -0,0 +1,105 @@
+var path = require('path');
+var url = require('url');
+var fs = require('fs');
+
+import * as tl from 'vsts-task-lib/task';
+
+import * as models from 'artifact-engine/Models';
+import * as engine from 'artifact-engine/Engine';
+import * as providers from 'artifact-engine/Providers';
+import * as webHandlers from 'artifact-engine/Providers/typed-rest-client/Handlers';
+
+var taskJson = require('./task.json');
+
+tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+const area: string = 'DownloadFileShareArtifacts';
+
+function getDefaultProps() {
+    var hostType = (tl.getVariable('SYSTEM.HOSTTYPE') || "").toLowerCase();
+    return {
+        hostType: hostType,
+        definitionName: '[NonEmail:' + (hostType === 'release' ? tl.getVariable('RELEASE.DEFINITIONNAME') : tl.getVariable('BUILD.DEFINITIONNAME')) + ']',
+        processId: hostType === 'release' ? tl.getVariable('RELEASE.RELEASEID') : tl.getVariable('BUILD.BUILDID'),
+        processUrl: hostType === 'release' ? tl.getVariable('RELEASE.RELEASEWEBURL') : (tl.getVariable('SYSTEM.TEAMFOUNDATIONSERVERURI') + tl.getVariable('SYSTEM.TEAMPROJECT') + '/_build?buildId=' + tl.getVariable('BUILD.BUILDID')),
+        taskDisplayName: tl.getVariable('TASK.DISPLAYNAME'),
+        jobid: tl.getVariable('SYSTEM.JOBID'),
+        agentVersion: tl.getVariable('AGENT.VERSION'),
+        agentOS: tl.getVariable('AGENT.OS'),
+        agentName: tl.getVariable('AGENT.NAME'),
+        version: taskJson.version
+    };
+}
+
+function publishEvent(feature, properties: any): void {
+    try {
+        var splitVersion = (process.env.AGENT_VERSION || '').split('.');
+        var major = parseInt(splitVersion[0] || '0');
+        var minor = parseInt(splitVersion[1] || '0');
+        let telemetry = '';
+        if (major > 2 || (major == 2 && minor >= 120)) {
+            telemetry = `##vso[telemetry.publish area=${area};feature=${feature}]${JSON.stringify(Object.assign(getDefaultProps(), properties))}`;
+        }
+        else {
+            if (feature === 'reliability') {
+                let reliabilityData = properties;
+                telemetry = "##vso[task.logissue type=error;code=" + reliabilityData.issueType + ";agentVersion=" + tl.getVariable('Agent.Version') + ";taskId=" + area + "-" + JSON.stringify(taskJson.version) + ";]" + reliabilityData.errorMessage
+            }
+        }
+        console.log(telemetry);;
+    }
+    catch (err) {
+        tl.warning("Failed to log telemetry, error: " + err);
+    }
+}
+
+async function main(): Promise<void> {
+    var promise = new Promise<void>(async (resolve, reject) => {
```

Comment:
This is a new task, so adopt new standards. let or const where applicable instead of var.

Reply:
done.